### PR TITLE
[GUILD] Typescript Island

### PR DESCRIPTION
**Pull request by jelledc**
### Description

Converted `Island` and `IslandGroup` components to typescript.

### CODE REVIEW

File: src/components/island/Island.tsx [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
 };
 
-/** @type {React.ComponentType<any>} */
-const Island = (props) => {
-  const { children, className, color, size, onClick, ...others } = props;
+interface IslandProps extends Omit<BoxProps, 'size'> {
+  /** A class name for the button to give custom styles. */
+  className?: string;
+  /** The color for the Island background and border */
+  color?: 'neutral' | 'mint' | 'violet' | 'ruby' | 'gold' | 'aqua' | 'white';
```
Comment (KristofColpaert) [inline src/components/island/Island.tsx:18]: You could use the constant colors here

File: src/components/island/IslandGroup.tsx [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
-import PropTypes from 'prop-types';
+import React, { forwardRef, isValidElement, ReactElement } from 'react';
+import { GenericComponent } from '../../@types/types';
+
 import Box, { pickBoxProps } from '../box';
+import { BoxProps } from '../box/Box';
+
+interface IslandGroupProps extends Omit<BoxProps, 'size'> {
+  /** The color of the child Islands */
+  color?: 'neutral' | 'mint' | 'violet' | 'ruby' | 'gold' | 'aqua' | 'white';
```
Comment (KristofColpaert) [inline src/components/island/IslandGroup.tsx:9]: You could use the constant colors here

File: src/components/island/Island.tsx (hunk 1 above)
Comment (jelledc) [reply src/components/island/Island.tsx:18]: Changed 👍

File: src/components/island/IslandGroup.tsx (hunk 2 above)
Comment (jelledc) [reply src/components/island/IslandGroup.tsx:9]: Changed 👍

**Review by qubis741 — APPROVED** [review]

File: src/components/island/Island.tsx [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
 
-/** @type {React.ComponentType<any>} */
-const Island = (props) => {
-  const { children, className, color, size, onClick, ...others } = props;
+export interface IslandProps extends Omit<BoxProps, 'size'> {
+  /** A class name for the button to give custom styles. */
+  className?: string;
+  /** The color for the Island background and border */
+  color?: Exclude<typeof COLORS[number], 'teal'> | 'white';
+  /** Te size of the Island component */
```
Comment (lorgan3) [inline src/components/island/Island.tsx:20]:
```suggestion
  /** The size of the Island component */
```

**Review by lorgan3 — APPROVED** [review]

**Review by KristofColpaert — APPROVED** [review]

**Comment by qubis741** [comment]
Manually checked :white_check_mark:

**Pull request merged**